### ISSUE TITLE
Annotations doesn't seems to work when importing code from a file into a codeblock

### ISSUE THREAD

**Issue by aromko** [code-hike/codehike#439]
Hey,
I'm trying to use annotations when importing code from a file. It works in the old version but in the new version it seems that I'm missing something.

I tried this: 

```
<CH.Code style={{ height: 500 }}>

`tsx TabsActionExample.tsx use-v1
// !focus(14:22)
!from ./_components/TabsActionExample.tsx
`

`tsx Link.tsx
// from ./_components/Link.tsx
`

</CH.Code>
```

This leads to: 
![Bildschirmfoto 2024-08-30 um 10 19 00](https://github.com/user-attachments/assets/7c41d81d-e263-44cb-b1ba-b49545a4619e)

When I remove `// !focus(14:22)` the code is loaded.

What am I doing wrong?

**Comment by pomber** [comment]
you are using v1.0.0, right? yes, currently you can only have the `!from` inside the codeblock. you can put annotations inside the imported file if it helps, but the idea is to make it more flexible soon, allowing annotations, extra code, extra `from`s. 

related #401

**Comment by aromko** [comment]
Yeah. sorry I'm trying to upgrade to the new version.

**Comment by pomber** [comment]
no problem, let me know any other issue you have

**Comment by aromko** [comment]
So you mean I can use the annotations in the .tsx files? e.g 

```
const App = () => (
  // !focus(14:22)
  <MarigoldProvider theme={theme} className="h-[340px]">
    <Tabs defaultActiveTab="description">
...
```

That doesn't seems to work and I think I don't want it into my components :-D So I need to wait until it is supported :-/

**Comment by aromko** [comment]
Are they plans to support `!from` outside a codeblock like in the old version in the near future?

**Comment by pomber** [comment]
> @pomber Are they plans to support `!from` outside a codeblock like in the old version in the near future?

you mean annotations outside of the imported files right? if not, provide an example of what you need. I'll work on some `!from` improvements soon.

**Comment by aromko** [comment]
Yes, exactly. Nice to hear. I'm waiting for it to continue migration 😁

**Comment by aromko** [comment]
Hey @pomber. How is the status here? Still waiting for this feature to finally migrate to the latest version 🙂

**Comment by chailandau** [comment]
Hey @pomber we are also running into this -- any updates on this?

**Comment by chailandau** [comment]
For folks having similar issues, using this as a stopgap and its working great -- hopefully it helps others as well.

```
// src/components/codehike/annotations/utils.ts

import type { RawCode } from "codehike/code";

export function getFocusInfo(codeblock: RawCode): string {
  // Handle both formats:
  // - focus=5:9 (range)
  // - focus=17 (single line)

  const focusMatch = /.*focus=(\d+(?::\d+)?)/.exec(codeblock.meta);
  if (!focusMatch) return codeblock.value;

  const focusRange = focusMatch[1];
  const focusComment = `# !focus(${focusRange})\n`;
  return focusComment + codeblock.value;
}
```

```
// src/components/codehike/code/index.tsx

...
import { getFocusInfo } from "@/components/codehike/annotations/utils";

// we are processing the focus comment as part of the code snippet now, so it works
 return (
    <CodeSnippet
      code={getFocusInfo(codeblock)}
    />
...
```
